packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx

```diff
@@ -351,25 +379,29 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
   );
 
   const AttachmentButton = useCallback(() => {
+    /* @conditional-compile-remove(attachment-upload) */
     if (!attachmentOptions?.uploadOptions?.handleAttachmentSelection) {
       return null;
     }
+    /* @conditional-compile-remove(attachment-upload) */
     return (
       <AttachmentUploadButton
         supportedMediaTypes={attachmentOptions?.uploadOptions?.supportedMediaTypes}
         disableMultipleUploads={attachmentOptions?.uploadOptions?.disableMultipleUploads}
         onChange={attachmentUploadButtonOnChange}
       />
     );
+    return <></>;
   }, [
     attachmentOptions?.uploadOptions?.handleAttachmentSelection,
     attachmentOptions?.uploadOptions?.supportedMediaTypes,
     attachmentOptions?.uploadOptions?.disableMultipleUploads,
+    /* @conditional-compile-remove(attachment-upload) */
     attachmentUploadButtonOnChange
   ]);
 
   /* @conditional-compile-remove(attachment-upload) */
-  const attachmentsWithProgress = useSelector(attachmentUploadsSelector).attachments;
+  const attachmentsWithProgress = uploads?.map((v) => v.metadata);
```

Comment:
please use `useMemo` as otherwise we will call this map on every re-render and update `attachmentsWithProgress` which will cause re-renders for all other components (message thread, send box...)